### PR TITLE
bg/LWPMIOPEN-192: Integrate CK's batch norm forward training into non-tunable MIOpen solver 

### PR DESCRIPTION
**Pull request by bghimireamd**
* add CK's Batch Norm Forward Training solver for F16,  F32, F64, BF16 types
* NHWC Layout
* GTest added

### CODE REVIEW

**Comment by junliume** [comment]
@bghimireamd the following test is not passing
```
 test_bn_spatial_nhwc_test

**Comment by CAHEK7** [comment]
Could you please apply #2379

File: test/gtest/bn_test_data.hpp [hunk 1]
```diff
@@ -226,3 +226,106 @@ struct BNInferTestData : public BNTestData<XDataType, YDataType, TConfig>
...
+    }
+
+    void InitTensorsWithRandValue()
+    {
+        std::random_device rd{};
+        std::mt19937 gen{rd()};
+        std::uniform_int_distribution<> d{0, 100};
+        auto gen_value = [&](auto...) {
+            return 1e-2 * static_cast<ScaleDataType>(d(gen)) * ((d(gen) % 2 == 1) ? -1 : 1);
+        };
```
Comment (CAHEK7) [inline test/gtest/bn_test_data.hpp]:
```suggestion
        auto gen_value = [](auto...) {
            return prng::gen_descreet_uniform_sign(static_cast<ScaleDataType>(1e-2), 100);
        };
```
See #2397 for details.
Also #include <random> at line 28 should be replaced with #include "random.hpp".

File: test/gtest/bn_test_data.hpp [hunk 2]
```diff
@@ -226,3 +226,106 @@ struct BNInferTestData : public BNTestData<XDataType, YDataType, TConfig>
...
+        scale.generate(gen_value);
+        shift.generate(gen_value);
+
+        auto gen_var = [&](auto...) { return 1e-2 * (static_cast<MeanVarDataType>(d(gen)) + 1); };
```
Comment (CAHEK7) [inline test/gtest/bn_test_data.hpp]:
```suggestion
        auto gen_var = [](auto...) {
            return static_cast<T>(1e-2) * static_cast<MeanVarDataType>(prng::gen_0_to_B(100) + 1);
        };
```

**Review by CAHEK7 — CHANGES_REQUESTED** [review]

**Review by CAHEK7 — APPROVED** [review]

**Review by junliume — APPROVED** [review]

**Pull request merged**